src/fixupReactRouter.js

```diff
@@ -0,0 +1,52 @@
+
+function ensureArray(maybeArray) {
+  if (!maybeArray) {
+    return [];
+  } else {
+    if (Array.isArray(maybeArray)) {
+      return maybeArray;
+    } else {
+      return [maybeArray];
+    }
+  }
+}
+
+function isReactRouterish(type) {
+  return type && (
+    type.displayName === 'Router' ||
+    type.name === 'Router' // In case Ryan and Michael embrace ES6 classes
+  );
+}
+
+function componentsToForceUpdate(routes) {
```

Comment:
Let’s rename to `extractComponents`. We’re not exactly force updating them anyway.